src/comps/Deposit.tsx

```diff
@@ -511,11 +547,27 @@ const DepositFlow = () => {
     },
     [setTxId],
   );
+  const { addresses } = useAtomValue(walletInfoAtom);
+  const btcAddress = addresses.payment?.address;
+  const { data: btcBalance } = useQuery({
+    queryKey: ["btcBalance", btcAddress],
+    queryFn: async () => {
+      if (!btcAddress) {
+        return 0;
+      }
+      return await getBtcBalance(btcAddress);
+    },
+    enabled: !!btcAddress,
+  });
   const renderStep = () => {
     switch (step) {
       case DEPOSIT_STEP.AMOUNT:
         return (
-          <DepositFlowAmount setAmount={setAmount} setStep={handleUpdateStep} />
+          <DepositFlowAmount
+            btcBalance={btcBalance || 0}
```

Comment:
IIUC defaulting to `0` means that if we are having troubles reaching out to the mempool api, users will not be able to use the bridge. If so, should we use a `MAX_MONEY` default instead (or disable validation if we cannot get the balance)? Alternatively, we could change the check wrt the user balance to be just a fyi kind of thing, not blocking the flow -- to that if we/api mess up the balance computation, the user can still go ahead with the process.